src/storage/lock_manager/deadlock.rs

```diff
@@ -6,26 +6,37 @@ use super::util::extract_physical_timestamp;
 use super::waiter_manager::Scheduler as WaiterMgrScheduler;
 use super::{Error, Lock, Result};
 use crate::pd::{RpcClient, INVALID_ID};
+use crate::raftstore::coprocessor::{Coprocessor, CoprocessorHost, ObserverContext, RoleObserver};
 use crate::server::resolve::StoreAddrResolver;
 use crate::tikv_util::collections::HashMap;
 use crate::tikv_util::future::paired_future_callback;
 use crate::tikv_util::security::SecurityManager;
-use crate::tikv_util::worker::{FutureRunnable, FutureScheduler, Stopped};
+use crate::tikv_util::worker::{FutureRunnable, FutureScheduler, FutureWorker, Stopped};
 use futures::{Future, Sink, Stream};
 use grpcio::{
     self, DuplexSink, RequestStream, RpcContext, RpcStatus, RpcStatusCode, UnarySink, WriteFlags,
 };
 use kvproto::deadlock::*;
 use kvproto::deadlock_grpc;
+use kvproto::metapb::Region;
 use protobuf::RepeatedField;
+use raft::StateRole;
 use std::cell::RefCell;
 use std::fmt::{self, Display, Formatter};
 use std::rc::Rc;
 use std::sync::Arc;
-use std::time::{Duration, Instant};
+use std::time::Duration;
 use tokio_core::reactor::Handle;
 use tokio_timer::Interval;
 
+/// The leader of the region containing the LEADER_KEY is the leader of deadlock detector.
+const LEADER_KEY: &[u8] = b"";
+
+/// Returns true if the region containing the LEADER_KEY.
+fn is_leader_region(region: &'_ Region) -> bool {
+    region.get_start_key() <= LEADER_KEY && LEADER_KEY < region.get_end_key()
+}
+
 // 2 mins
 const TXN_DETECT_INFO_TTL: u64 = 120000;
```

Comment:
This value can be reduced?

Reply:
See https://github.com/tikv/tikv/pull/5028/files#diff-fa2bb0fa5ba566c8685b2a159a41ada6R291
If the performance is good, the timer will be removed. If not, I will file a PR to change it.